.github/workflows/test.yaml

```diff
@@ -1,87 +1,110 @@
-
-jobs:
-  test:
-    strategy:
-      fail-fast: false # finalize testing of all targets even if one failed
-      matrix:
-        include:
-          - name: "MacOS x86"
-            runner: macos-12-large
-          - name: "MacOS arm64"
-            runner: [self-hosted, macOS, ARM64]
-          - name: "Linux x86"
-            runner: matterlabs-ci-runner
-            image: matterlabs/llvm_runner:ubuntu22-llvm17-latest
-            target: "x86_64-unknown-linux-musl"
-          - name: "Linux ARM64"
-            runner: matterlabs-ci-runner-arm
-            image: matterlabs/llvm_runner:ubuntu22-llvm17-latest
-            target: "aarch64-unknown-linux-musl"
-          - name: "Windows"
-            runner: windows-2022-github-hosted-16core
-    runs-on: ${{ matrix.runner }}
-    container:
-      image: ${{ matrix.image || '' }} # Special workaround to allow matrix builds with optional container
-    name: ${{ matrix.name }}
-    steps:
-      - name: Checkout source
-        uses: actions/checkout@v4
-
-      - name: Prepare Windows env
-        if: runner.os == 'Windows'
-        uses: ./.github/actions/prepare-msys
-
-      - name: Build LLVM
-        uses: ./.github/actions/build-llvm
-
-      - name: Build zksolc
-        uses: ./.github/actions/build
-        with:
-          target: ${{ matrix.target }}
-
-      - name: Install solc compiler
-        uses: ./.github/actions/install-solc
-        with:
-          solc-version: ${{ github.event.inputs.solc-version }}
-
-      - name: Run tests
-        uses: ./.github/actions/unit-tests
-        with:
-          target: ${{ matrix.target || '' }}
-
-      - name: Run CLI tests
-        uses: ./.github/actions/cli-tests
-        with:
-          use-prebuilt-zksolc: false # always build zksolc from sources for general PR testing
-          zksolc-version: ${{ github.event.inputs.zksolc-version || '' }}
+name: Tests
+
+# Execute workflow for each PR and with each merge to the trunk
+on:
+  workflow_dispatch:
+    inputs:
+      zksolc-version:
+        type: string
+        description: "Prebuilt zksolc version to use, (repo: https://github.com/matter-labs/zksolc-bin/releases)"
+        required: true
+        default: "1.4.1"
+      solc-version:
+        type: string
+        description: "Prebuilt solc version, (repo: https://github.com/matter-labs/era-solidity/releases)"
+        required: true
+        default: "0.8.25-1.0.0"
+  pull_request:
+    paths-ignore:
+      - '.gitignore'
+      - '**/*.md'
+  push:
+    branches:
+      - main
+
+# Cancel the workflow if any new changes pushed to a feature branch or the trunk
+concurrency:
+  group: ${{ github.workflow }}-${{ github.event.pull_request.number || github.ref }}
+  cancel-in-progress: true
+
+jobs:
+
+  # Check for cargo licensing issues
+  cargo-license:
+    uses: matter-labs/era-compiler-tester/.github/workflows/cargo-license.yaml@aba-cpr-1691-set-up-a-pipeline-to-apply-all-ci-changes-to-all-repos

```

Comment:
Let's use another repo for CI.
Using `era-compiler-tester` here adds an inverted dependency tree on top of the Rust one, which is architectually incorrect.

Reply:
What about the repo name? I'm thinking something like `era-compiler-ci`.